scripts/pi-hole/php/auth.php

```diff
@@ -73,7 +73,7 @@ function check_cors() {
     $server_host = str_replace(array("[","]"), array("",""), $server_host);
 
     if(isset($_SERVER['HTTP_HOST']) && !in_array($server_host, $AUTHORIZED_HOSTNAMES)) {
-        log_and_die("Failed Host Check: " . $server_host .' vs '. join(', ', $AUTHORIZED_HOSTNAMES));
+        log_and_die("Failed Host Check: " . $server_host .' vs '. join(', ', htmlspecialchars($AUTHORIZED_HOSTNAMES)));
```

Comment:
`$AUTHORIZED_HOSTNAMES` is an array of values.
`join(', ', $AUTHORIZED_HOSTNAMES)` is a string.

You should do:

```php
        log_and_die("Failed Host Check: " . $server_host .' vs '. ', htmlspecialchars(join(', $AUTHORIZED_HOSTNAMES)));
```